### ISSUE TITLE
Unexpected Pauli composition result (sign reversed)

### ISSUE THREAD

**Issue by we-taper** [Qiskit/qiskit#8095]
Compose of two Pauli operators gives unexpected sign:

```python
from qiskit.quantum_info import Pauli
import itertools
for a, b in itertools.product('XYZ', repeat=2):
    print(f'{a} {b} -> {Pauli(a) & Pauli(b)}')
```
gives:
```
X X -> I
X Y -> -iZ
X Z -> iY
Y X -> iZ
Y Y -> I
Y Z -> -iX
Z X -> -iY
Z Y -> iX
Z Z -> I
```
when we naively expects, for example XY = iZ. This is also confirmed by:

```python
(Pauli('X').to_matrix() @ Pauli('Y').to_matrix() == Pauli('iZ').to_matrix()).all()
# True
```

Is this reverse of sign intentionally? It seems difficult for me to find a documentation mentioning this as well.

Thanks!

**Comment by jakelishman** [comment]
Yes, that's how we define difference between "compose" and "dot" - compose (`&`) means "apply the left one, then apply the right one", whereas dot (`@`) is the mathematical matrix multiplication.

You're right though that the documentation isn't necessarily easy to find - you'd need to know that `&` is the syntactic sugar we decided to use for [`Operator.compose`](https://qiskit.org/documentation/stubs/qiskit.quantum_info.Operator.compose.html).  We do say it, but I do recognise it's not the best layout.  We've got a big internal effort to work on how we present our documentation, but it might take a little while to bear fruit.
